lib/src/posthog_widget_widget.dart

```diff
@@ -65,14 +66,11 @@ class PostHogWidgetState extends State<PostHogWidget> {
   }
 
   Future<void> _generateSnapshot() async {
-    final isSessionReplayActive =
-        await _nativeCommunicator?.isSessionReplayActive() ?? false;
-    if (!isSessionReplayActive) {
-      return;
-    }
-
+    // Ensure no asynchronous calls occur before this function,
```

Comment:
calling `isSessionReplayActive` is important even if its inside `captureScreenshot` because we don't want to pay the performance price of taking the screenshot if session replay is inactive

Reply:
yes, I put this in 

https://github.com/PostHog/posthog-flutter/blob/25a3690d6fe4cb5d53516c7b70161fffaabd106e/lib/src/replay/screenshot/screenshot_capturer.dart#L111-L117

what do you think?